enhancements/workload-partitioning/pao-render.md

```diff
@@ -0,0 +1,190 @@
+   sponsor the process.
+1. **Merge at each milestone.** Merge when the design is able to transition to a
+   new status (provisional, implementable, implemented, etc.). View anything
+   marked as `provisional` as an idea worth exploring in the future, but not
+   accepted as ready to execute. Anything marked as `implementable` should
+   clearly communicate how an enhancement is coded up and delivered. If an
+   enhancement describes a new deployment topology or platform, include a
+   logical description for the deployment, and how it handles the unique aspects
+   of the platform. Aim for single topic PRs to keep discussions focused. If you
+   disagree with what is already in a document, open a new PR with suggested
+   changes.
+1. **Keep all required headers.** If a section does not apply to an
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+The `Metadata` section above is intended to support the creation of tooling
+around the enhancement process.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Performance addon operator is a day 2 optional operator installed via OLM. However the [Workload partitioning](management-workload-partitioning.md) feature needs to have some configuration ready at the installation time when PAO is not yet ready.
+This feature is about adding a `render` mode to PAO that would allow an admin to pre-compute all the Openshift manifests that are needed in a way that prevents any typing mistakes from happening.
+These generated manifests can then be passed to the installer and later taken over by the PAO reconcile loop. 
+
+## Motivation
+
+There are many pieces that must be configured just right for the latency and resource management to work properly. PAO addresses that for day 2 latency tuning, but there was no way to use PAO's knowledge at the install time.
+
+### Goals
+
+1) PAO can execute outside of a cluster via podman/docker, consume user manifests from a directory and generate additional manifests to another (or the same) directory
+1) PAO takes all PerformanceProfile yamls from the input directory and generates all the usual manifests as yaml files to the output directory
+1) When a command line option `--enable-workload-partitioning` is passed to PAO in render mode, it also generates the necessary configuration for the `management` partition to work properly
+1) PAO at day 2 must not make any changes to the existing generated objects it owns
+1) PAO sets the ownership references on the generated yamls so the generated files are properly linked to the PerformanceProfiles
```

Comment:
Lines 89-92 are implementation details. They could go into the general proposal section around line 101 with a little bit of text to tie them together into a paragraph, or they could go down to line 110.

Reply:
Well I would treat these as acceptance criteria. Because it is the "API contract" of the flow.

Reply:
I just think you're being too specific, too early.

We usually try to set enough context about the requirements that a reviewer reading the details later can evaluate the choices we're making against the goals to see if they match or if we have gaps. If we set as a goal to implement something a specific way, with specific interfaces, then it's like we're not asking for feedback on the design.

So, for lines 89 and 90 we could say something like

* The PAO running outside of a cluster can provide input to the OpenShift installer to enable workload partitioning during cluster deployment in a way that does not require changes to the OpenShift installer.

Then for 91 maybe

* The configuration from the PAO that is passed to the installer is compatible with the configuration the PAO generates when running inside the cluster.

And after re-reading 92 with those proposed changes, I think it's OK to keep it as it is.